app/src/pages/inside/testCaseLibraryPage/createTestCaseModal/useCreateTestCase.ts

```diff
@@ -14,149 +14,81 @@
-        linkToRequirements: payload.linkToRequirements,
-        manualScenarioType: payload.manualScenarioType,
-        preconditions: {
-          value: payload.precondition,
-          attachments: payload.preconditionAttachments ?? [],
-        },
-      };
-
-      const manualScenario: ManualScenarioDto =
-        payload.manualScenarioType === 'TEXT'
-          ? {
-              ...commonData,
-              instructions: payload.instructions,
-              expectedResult: payload.expectedResult,
-              attachments: payload.textAttachments ?? [],
-            }
-          : {
-              ...commonData,
-              steps: Object.values(payload?.steps ?? {}).map((step) => ({
-                instructions: step.instructions,
-                expectedResult: step.expectedResult,
-                attachments: step.attachments ?? [],
-              })),
-            };
-
-      await fetch(URLS.testCase(projectKey), {
-        method: 'post',
-        data: {
-          description: payload.description,
-          name: payload.name,
-          testFolderId: folderId,
-          priority: payload.priority.toUpperCase(),
-          manualScenario,
-        },
-      });
-
-      dispatch(hideModalAction());
-      dispatch(
-        showSuccessNotification({
-          messageId: 'testCaseCreatedSuccess',
-        }),
-      );
-      dispatch(getTestCasesAction({ testFolderId: folderId }));
-    } catch (error: unknown) {
-      if (error instanceof Error && error?.message?.includes('tms_test_case_name_folder_unique')) {
-        throw new SubmissionError({
-          name: formatMessage(messages.duplicateTestCaseName),
+  const createTestCase = useCallback(
+    async (payload: CreateTestCaseFormData) => {
+      try {
+        showSpinner();
+
+        let folderId: number;
+
+        if (isString(payload.folder)) {
+          const folder = await createFolder(projectKey, payload.folder);
+          dispatch(createFoldersSuccessAction({ ...folder, countOfTestCases: 0 }));
+          folderId = folder.id;
+        } else {
+          folderId = payload.folder?.id;
+        }
+
+        const manualScenario = buildManualScenario(payload);
+
+        await fetch(URLS.testCase(projectKey), {
+          method: 'post',
+          data: {
+            description: payload.description,
+            name: payload.name,
+            testFolderId: folderId,
+            priority: payload.priority?.toUpperCase(),
+            manualScenario,
+          },
         });
-      } else {
+
+        dispatch(hideModalAction());
         dispatch(
-          showErrorNotification({
-            messageId: 'testCaseCreationFailed',
+          showSuccessNotification({
+            messageId: 'testCaseCreatedSuccess',
           }),
         );
+        dispatch(getTestCasesAction({ testFolderId: folderId }));
+      } catch (error: unknown) {
+        try {
+          handleTestCaseError(error, formatMessage, messages.duplicateTestCaseName);
```

Comment:
Can we move `messages.duplicateTestCaseName` to the function?
```suggestion
          handleTestCaseError(error, formatMessage);
```